### DISCUSSION TITLE
Use result of `func` paramter in handler.

### DISCUSSION THREAD

**Discussion by byehack** [eternnoir/pyTelegramBotAPI#1731] [Ideas]
```py
def fn(message):
    return ", Word!"

@bot.message_handler(func=fn)
async def hndler(message, func_result):
    print("Hello"+func_result)
```

**Comment by coder2020official** [comment]
Here are my points about it:
- Func should return True/False. If False was returned, then handler won't run. If True is returned, then handler will run. What is the point of getting True always(you can return other values but purpose is to create lambda filters)?
- If you need to pass something to handlers, I'd recommend taking a look at middlewares

**Reply by byehack** [reply]
> * (you can return other values but purpose is to create lambda filters)

Some times the filter function is longer than is to fit lambda.

```py
def fn(message):
    if is_text_is_link(message.text):
        content =  download(message.text).content
        filtered_content = filter_content(content)
        if is_ok(filtered_content): return True

@bot.message_handler(func=fn)
async def hndler(message):
    content =  download(message.text).content
    filtered_content = filter_content(content)
    bot.send_message(filtered_content)
```

by accept func paremeter result in handler:

```py
def fn(message):
    if is_text_is_link(message.text):
        content =  download(message.text).content
        filtered_content = filter_content(content)
        if is_ok(filtered_content): return filtered_content

@bot.message_handler(func=fn)
async def hndler(message, func_result):
    bot.send_message(func_result)
```
this avoid duplicate processing. (more efficient)



> * Func should return True/False.

No, Func result most not always be `bool`, it can return type of `Any` according to here:
https://github.com/eternnoir/pyTelegramBotAPI/blob/b841fc10eddc99c819f8a3be0684b605d520e921/telebot/__init__.py#L5859



I think here is should try to start supporting it.

https://github.com/eternnoir/pyTelegramBotAPI/blob/b841fc10eddc99c819f8a3be0684b605d520e921/telebot/__init__.py#L5929

**Comment by coder2020official** [comment]
The purpose of it is to return true false eventhough it allows to pass other data types; Take a look at custom filters. The purpose of custom filters is to create such "long" filter functions.

**Reply by byehack** [reply]
> The purpose of custom filters is to create such "long" filter functions.


:/ still duplicate `download` and duplicate processing.
```py
class cust(telebot.custom_filters.SimpleCustomFilter):
    key='is_my_link'
    @staticmethod
    def check(message: telebot.types.Message):
        if is_text_is_link(message.text):
            content =  download(message.text).content
            filtered_content = filter_content(content)
            if is_ok(filtered_content): return True
	
bot.add_custom_filter(cust())

@bot.message_handler(is_my_link=True)
def hndler(message):
    content =  download(message.text).content
    filtered_content = filter_content(content)
    bot.send_message(filtered_content)
```

**Comment by coder2020official** [comment]
Then create a class-based middleware, create a check there , and add download to data.

**Reply by byehack** [reply]
:/ why just not use clear code and simple way?

**Reply by byehack** [reply]
The purpose can change.
and also it is backward compatible!

**Reply by coder2020official** [reply]
Using middlewares you can achieve that easily;
Do it through middlewares

**Comment by Cub11k** [comment]
@byehack you totally misunderstood the concept :)
Any arguments in the decorator are filters, they are supposed to define whether to execute this handler or skip it and try others. Their purpose won't ever change

To pass any extra parameters to the handler you should use middleware
